### PR TITLE
Backporting Transition to JSON logging to V1

### PR DESCRIPTION
**Pull request by davidmrdavid**
Tackles: https://github.com/Azure/azure-functions-durable-extension/issues/1733
Backports: https://github.com/Azure/azure-functions-durable-extension/pull/1721

### Issue describing the changes in this PR

resolves https://github.com/Azure/azure-functions-durable-extension/issues/1733

### CODE REVIEW

File: release_notes.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+## New Features
+- Improved supportability logs on Linux Dedicated plans (https://github.com/Azure/azure-functions-durable-extension/pull/1721)
+
+## Bug fixes
+- Fix issue with local RPC endpoint used by non-.NET languages on Windows apps (#1800)
+
+## Breaking Changes
```
Comment (davidmrdavid) [inline release_notes.md:7]: I'll remove this

File: src/WebJobs.Extensions.DurableTask/EventSourceListener.cs [hunk 2]
```diff
@@ -39,10 +39,8 @@ public EventSourceListener(LinuxAppServiceLogger logger, bool enableVerbose, End
         /// <param name="eventSource">An instance of EventSource.</param>
         protected override void OnEventSourceCreated(EventSource eventSource)
         {
-            // "7DA4779A-152E-44A2-A6F2-F80D991A5BEE" is the old DurableTask-Core event source,
-            // so we provide extra logic to ignore it.
-            if ((eventSource.Name == "DurableTask-Core"
-                  && eventSource.Guid != new Guid("7DA4779A-152E-44A2-A6F2-F80D991A5BEE")) ||
+
+            if (eventSource.Name == "DurableTask-Core" ||
```
Comment (davidmrdavid) [inline src/WebJobs.Extensions.DurableTask/EventSourceListener.cs:43]: So, for the code in `dev`, we explicitly filter out the "DurableTask-Core" provider of GUID "7DA4779A-152E-44A2-A6F2-F80D991A5BEE" since it "is the old DurableTask-Core event source". From my testing, in V1, it is also the _only_ DurableTask-Core event source, so filtering it out means we get no Core logs. Does that ring true to you all?

**Comment by davidmrdavid** [comment]
This still needs E2E testing, but I'll complete that prior to a merge

**Comment by davidmrdavid** [comment]
I seem to recall E2E testing this, but it's been a while, so I can do it again. Still, I'd appreciate a conditional approval :)

File: src/WebJobs.Extensions.DurableTask/DurableTaskOptions.cs [hunk 3]
```diff
@@ -34,7 +34,6 @@ public class DurableTaskOptions
         /// </summary>
         public TraceOptions Tracing { get; set; } = new TraceOptions();
 
-
```
Comment (ConnorMcMahon) [inline src/WebJobs.Extensions.DurableTask/DurableTaskOptions.cs:37]: We may want to revert this change.

File: src/WebJobs.Extensions.DurableTask/EventSourceListener.cs (hunk 2 above)
Comment (ConnorMcMahon) [reply src/WebJobs.Extensions.DurableTask/EventSourceListener.cs:43]: I don't believe it is the only DurableTask-Core implementation, because we update v1's version of DurableTask.AzureStorage to get the latest changes. That will inherently increase our DurableTask.Core version to one that has the new event source.

**Review by ConnorMcMahon — CHANGES_REQUESTED** [review]
I think we want to filter out the old ones like we do in dev, but double check the version of DurableTask.Core to see if it is the version that has both event sources.

File: src/WebJobs.Extensions.DurableTask/DurableTaskOptions.cs (hunk 3 above)
Comment (davidmrdavid) [reply src/WebJobs.Extensions.DurableTask/DurableTaskOptions.cs:37]: Hmm, but I think the code wouldn't compile with two blank lines in a row, right?

Comment (ConnorMcMahon) [reply src/WebJobs.Extensions.DurableTask/DurableTaskOptions.cs:37]: Ah, that makes sense.

File: src/WebJobs.Extensions.DurableTask/EventSourceListener.cs (hunk 2 above)
Comment (davidmrdavid) [reply src/WebJobs.Extensions.DurableTask/EventSourceListener.cs]: I trust that's correct but somehow my tests were missing logs when I had that filter enabled. I'll add the filter back and try again, it's been a while since I validated this.

Comment (davidmrdavid) [reply src/WebJobs.Extensions.DurableTask/EventSourceListener.cs]: Yeah, in my tests, the filtere'd out GUID is the only ETW event with name "DurableTask-Core".
I'll sync with you about this internally early next week

**Comment by cgillum** [comment]
I'm not sure if this has been covered already in other PR comments, but I noticed that the v1 branch points to an old version of DurableTask.AzureStorage:

```xml
<PackageReference Include="Microsoft.Azure.DurableTask.AzureStorage" Version="1.7.7" />
```

Can we update this to point to the latest (currently **1.8.5**)?

File: src/WebJobs.Extensions.DurableTask/EventSourceListener.cs (hunk 2 above)
Comment (cgillum) [reply src/WebJobs.Extensions.DurableTask/EventSourceListener.cs]: We need to keep this filter as-is and change our DurableTask.AzureStorage dependency to the latest version.

Comment (davidmrdavid) [reply src/WebJobs.Extensions.DurableTask/EventSourceListener.cs]: @cgillum, yup, turned out that was the culprit :) . Thanks for pointing that out!

**Comment by ConnorMcMahon** [comment]
@davidmrdavid, we ready to do final validation on this with 1.8.5 of DurableTask.AzureStorage now that #1838 was merged?

**Comment by davidmrdavid** [comment]
@ConnorMcMahon, yes but I'll need to pull in some changes and re-push. I'm responding to feedback in a different PR, but I'll get back to this shortly, I'll send you a re-review request once this PR is ready to be validated once more.

File: test/Common/TestHelpers.cs [hunk 4]
```diff
@@ -111,6 +113,98 @@ public static string GetTaskHubNameFromTestName(string testName, bool enableExte
...
+                "EventPrimaryStampName",
+                "ProviderName",
+                "TaskName",
+                "EventId",
+                "EventTimestamp",
+                "Tenant",
+                "Pid",
+                "Tid",
+            };
+            List<string> keys = json.Properties().Select(p => p.Name).ToList();
```
Comment (cgillum) [inline test/Common/TestHelpers.cs:135]: nit: `HashSet<string>` would be more appropriate than `List<string>`. That or just do `json.TryGetValue(expectedKey, out _)` and avoid the extra data structure altogether.

File: test/Common/TestHelpers.cs [hunk 5]
```diff
@@ -111,6 +113,98 @@ public static string GetTaskHubNameFromTestName(string testName, bool enableExte
...
+            using (var fileStream = new FileStream(path, FileMode.Open, FileAccess.Read, FileShare.ReadWrite))
+            using (var streamReader = new StreamReader(fileStream, Encoding.Default))
+            {
+                List<string> file = new List<string>();
+                while (!streamReader.EndOfStream)
+                {
+                    file.Add(streamReader.ReadLine());
+                }
+
+                return file.ToArray();
```
Comment (cgillum) [inline test/Common/TestHelpers.cs:169]: nit: This isn't important for test code, but it would be more efficient just to return `file` rather than call `ToArray()` on it (which copies all the elements).

File: test/Common/TestHelpers.cs [hunk 6]
```diff
@@ -111,6 +113,98 @@ public static string GetTaskHubNameFromTestName(string testName, bool enableExte
...
+        public static bool IsValidJSONLog(JObject json)
+        {
+            List<string> expectedKeys = new List<string>
+            {
+                "EventStampName",
+                "EventPrimaryStampName",
+                "ProviderName",
+                "TaskName",
+                "EventId",
+                "EventTimestamp",
```
Comment (cgillum) [inline test/Common/TestHelpers.cs:130]: Probably out of scope for this PR, but with this change (or with changes that preceded it) will we get fractional seconds just like we do with Windows? Not being able to properly sort by timestamps has been killing me when doing debugging on Linux. :)

**Review by cgillum — APPROVED** [review]
No blockers from me - but it's probably best to get a final sign-off from Connor. Just a few small nits than you can ignore plus a question.

**Review by ConnorMcMahon — APPROVED** [review]
Chris raised some valid minor critiques, other than that, ready to merge.

File: test/Common/TestHelpers.cs (hunk 6 above)
Comment (davidmrdavid) [reply test/Common/TestHelpers.cs:130]: Ah yes, that's killing me as well. Unfortunately, the answer is no, and I explain more about why in here: https://github.com/Azure/azure-functions-durable-extension/issues/1843

Let's use that issue to track and discuss this work-item :)

**Comment by davidmrdavid** [comment]
@cgillum, thank you for your suggestions. I've applied them here, and also in V2 via this PR: https://github.com/Azure/azure-functions-durable-extension/pull/1844

**Comment by davidmrdavid** [comment]
Just e2e tested this again and it works. Merging!
